lib/openid_client_io.dart

```diff
@@ -46,36 +66,36 @@ class Authenticator {
   Future<void> cancel() async {
     final state = flow.authenticationUri.queryParameters['state'];
     _requestsByState[state!]?.completeError(Exception('Flow was cancelled'));
-    final server = await _requestServers.remove(port)!;
-    await server.close();
+    final server = await _requestServers.remove(port);
+    await server?.close();
   }
 
   static final Map<int, Future<HttpServer>> _requestServers = {};
-  static final Map<String, Completer<Map<String, String>>> _requestsByState =
-      {};
-
-  static Future<HttpServer> _startServer(int port) {
-    return _requestServers[port] ??=
-        (HttpServer.bind(InternetAddress.anyIPv4, port)
-          ..then((requestServer) async {
-            print('server started $port');
-            await for (var request in requestServer) {
-              print('request $request');
-              request.response.statusCode = 200;
-              request.response.headers.set('Content-type', 'text/html');
-              request.response.writeln('<html>'
-                  '<h1>You can now close this window</h1>'
-                  '<script>window.close();</script>'
-                  '</html>');
-              await request.response.close();
-              var result = request.requestedUri.queryParameters;
-
-              if (!result.containsKey('state')) continue;
-              await processResult(result);
-            }
-
-            await _requestServers.remove(port);
-          }));
+  static final Map<String, Completer<Map<String, String>>> _requestsByState = {};
+
+  static Future<HttpServer> _startServer(int port, String? htmlPage, String? redirectMessage) {
+    return _requestServers[port] ??= (HttpServer.bind(InternetAddress.anyIPv4, port)
+      ..then((requestServer) async {
+        log('Server started at port $port');
+        await for (var request in requestServer) {
+          request.response.statusCode = 200;
+          if (redirectMessage != null) {
+            request.response.headers.contentType = ContentType.html;
+            request.response.writeln(htmlPage ??
```

Comment:
htmlPage will never be written. from the Authenticator constructor assert say only one htmlPage or redirectMessage can be exist. But if the redirectMessage is null, this line will never be written.

Reply:
You are right. I will change it if I will have some spare time

Reply:
Sorry, my mistake. If you look to constructor redirectMessage won't be null, because it has default value. So htmlPage will work, but it can be better written. I will change it.

Reply:
Yeah you are right, I didnt see default value of redirectMessage. That my mistake too. 😅